library(ape)



####### Function to extract a list of species names for all clades in a phylogeny  ########
####### Based on the clade.members function from the package CAIC				   ########
####### (https://r-forge.r-project.org/projects/caic/) by David Orme               ########

all.clades <- function(phy) {

	foo <- function (x, phy, tip.labels = TRUE) 
			{
			if (class(phy) != "phylo") 
			stop("Phylogeny required")
			if (!(x %in% as.numeric(phy$edge))) 
			stop("Node not in range for phylogeny")

			while (any(x > min(phy$edge[,1])-1)) {

			maxx.loc <- which(x == max(x))
			maxx <- x[maxx.loc]
			x <- x[-maxx.loc]
			newx <- as.numeric(phy$edge[, 2][which(phy$edge[, 1] == maxx)])
			x <- c(x, newx)
			}

	x <- unique(x)
	if (tip.labels) {return(phy$tip.label[x])} else {return(x)}
	}


    nodes <- sort(unique(as.numeric(phy$edge)))
    nodes <- nodes[nodes > min(phy$edge[,1])-1]
    clade.list <- lapply(nodes, foo, phy=phy)
    
    return(clade.list)
    }


##### Beast file generator - returns Beast file with monophyletic groups as specified in the phylogeny
error<-0.001
PolytomyResolver <- function(phy, error=0.001, file.out="") {

sink(paste(file.out, ".xml", sep=""))

    clade.list <- all.clades(phy)
    group <- paste("group", 1:length(clade.list), sep = "")

cat('<?xml version="1.0" standalone="yes"?>\n')
cat("<!-- Generated by Polytomy Resolver v1.0                             -->\n")
cat("<!--   by Tyler Kuhn, Gavin Thomas and Arne Mooers                   -->\n")
cat("<!--                                                                 -->\n")
cat("<!--   For resolving polytomous nodes in a dated tree                -->\n")
cat("<!--    under a Birth-Death model                                    -->\n")
cat("<beast>","\n")
cat("\t<!-- The list of taxa analyse (can also include dates/ages).                 -->\n")
cat("\t<!-- ntax=",length(clade.list[[1]]),"                                                                 -->\n",sep="")
cat('\t<taxa id="taxa">\n')
for (j in 1:length(clade.list[[1]])) {
		cat("\t\t",'<taxon id="',clade.list[[1]][j],'"','/>',"\n", sep="")
	}
cat("\t",'</taxa>',"\n", sep="")

cat("\n")
cat('<!-- Constraints - Begin monophyly constraints -->\n')
cat("\n")
	
for (i in 1:length(clade.list)) {
        group <- paste("group", i, sep = "")
        taxa <- clade.list[[i]]
        cat("\t",'<taxa id=','"',group,'"','>',"\n",sep="")
		for (j in 1:length(clade.list[[i]])) {
    		cat("\t\t",'<taxon idref=','"',clade.list[[i]][j],'"','/>',"\n", sep="")
    	}
		cat("\t",'</taxa>',"\n", sep="")
    }
	
cat("\n")
cat('<!-- Constraints - End monophyly constraints -->\n')
cat("\n")
	
cat('\t<alignment id="alignment" dataType="nucleotide">\n')
for (i in 1:length(phy$tip.label)) {
		cat("\t\t<sequence>\n")
		cat("\t\t\t",'<taxon idref="',phy$tip.label[i],'"/>',"\n",sep="")
		cat("\t\t\tN\n")
		cat("\t\t</sequence>\n")
	}
cat("\t</alignment>\n")
cat("\t<!-- The unique patterns from 1 to end                                       -->\n")
cat("\t<!-- npatterns=0                                                             -->\n")
cat('\t<patterns id="patterns" from="1">\n')
cat('\t\t<alignment idref="alignment"/>\n')
cat("\t</patterns>\n")
cat("\t<!-- A prior on the distribution node heights defined given                  -->\n")
cat("\t<!-- a Birth-Death speciation process (Gernhard 2008).                       -->\n")
cat('\t<birthDeathModel id="birthDeath" units="substitutions">\n')
cat("\t\t<birthMinusDeathRate>\n")
cat('\t\t\t<parameter id="birthDeath.meanGrowthRate" value="1.0" lower="0.0" upper="Infinity"/>\n')
cat("\t</birthMinusDeathRate>\n")
cat("\t\t<relativeDeathRate>\n")
cat('\t\t\t<parameter id="birthDeath.relativeDeathRate" value="0.5" lower="0.0" upper="Infinity"/>\n')
cat("\t\t</relativeDeathRate>\n")
cat("\t</birthDeathModel>\n")
cat("\t<!-- This is a simple constant population size coalescent model              -->\n")
cat("\t<!-- that is used to generate an initial tree for the chain.                 -->\n")
cat('\t<constantSize id="initialDemo" units="substitutions">\n')
cat("\t\t<populationSize>\n")
cat('\t\t\t<parameter id="initialDemo.popSize" value="100.0"/>\n')
cat("\t\t</populationSize>\n")
cat("\t\t</constantSize>\n")
	
cat("\n")
cat('<!-- Constraints - Begin Start Tree -->\n')
cat("\n")
cat('\t<!-- The starting tree.                                                      -->\n')
cat('\t<newick id="startingTree">\n')
cat("\t\t",write.tree(phy),"\n",sep="")
cat('\t</newick>\n')
	
cat("\n")
cat('<!-- Constraints - End Start Tree -->\n')
cat("\n")
	
cat('\t<!-- Generate a tree model                                                   -->\n')
cat('\t<treeModel id="treeModel">\n')
cat('\t\t<coalescentTree idref="startingTree"/>\n')
cat('\t\t<rootHeight>\n')
cat('\t\t\t<parameter id="treeModel.rootHeight"/>\n')
cat('\t\t</rootHeight>\n')
cat('\t\t<nodeHeights internalNodes="true">\n')
cat('\t\t\t<parameter id="treeModel.internalNodeHeights"/>\n')
cat('\t\t</nodeHeights>\n')
cat('\t\t<nodeHeights internalNodes="true" rootNode="true">\n')
cat('\t\t\t<parameter id="treeModel.allInternalNodeHeights"/>\n')
cat('\t\t</nodeHeights>\n')
cat('\t</treeModel>\n')
cat('\t<!-- Generate a speciation likelihood for Yule or Birth Death                -->\n')
cat('\t<speciationLikelihood id="speciation">\n')
cat('\t\t<model>\n')
cat('\t\t\t<birthDeathModel idref="birthDeath"/>\n')
cat('\t\t</model>\n')
cat('\t\t<speciesTree>\n')
cat('\t\t\t<treeModel idref="treeModel"/>\n')
cat('\t\t</speciesTree>\n')
cat('\t</speciationLikelihood>\n')
cat('\t<!-- The strict clock (Uniform rates across branches)                        -->\n')
cat('\t<strictClockBranchRates id="branchRates">\n')
cat('\t\t<rate>\n')
cat('\t\t\t<parameter id="clock.rate" value="1.0"/>\n')
cat('\t\t</rate>\n')
cat('\t</strictClockBranchRates>\n')
cat('\t<!-- The HKY substitution model (Hasegawa, Kishino & Yano, 1985)             -->\n')
cat('\t<HKYModel id="hky">\n')
cat('\t\t<frequencies>\n')
cat('\t\t\t<frequencyModel dataType="nucleotide">\n')
cat('\t\t\t\t<frequencies>\n')
cat('\t\t\t\t\t<parameter id="frequencies" value="0.25 0.25 0.25 0.25"/>\n')
cat('\t\t\t\t</frequencies>\n')
cat('\t\t\t</frequencyModel>\n')
cat('\t\t</frequencies>\n')
cat('\t\t<kappa>\n')
cat('\t\t\t<parameter id="kappa" value="1.0" lower="0.0" upper="Infinity"/>\n')
cat('\t\t</kappa>\n')
cat('\t</HKYModel>\n')
cat('\t<!-- site model                                                              -->\n')
cat('\t<siteModel id="siteModel">\n')
cat('\t\t<substitutionModel>\n')
cat('\t\t\t<HKYModel idref="hky"/>\n')
cat('\t\t</substitutionModel>\n')
cat('\t</siteModel>\n')
cat('\t<treeLikelihood id="treeLikelihood" useAmbiguities="false">\n')
cat('\t\t<patterns idref="patterns"/>\n')
cat('\t\t<treeModel idref="treeModel"/>\n')
cat('\t\t<siteModel idref="siteModel"/>\n')
cat('\t\t<strictClockBranchRates idref="branchRates"/>\n')
cat('\t</treeLikelihood>\n')

cat("\n")
cat('\t<!-- Constraints - Begin Monophyly statistics -->\n')
cat("\n")
	for (i in 1:length(clade.list)) {
		group <- paste("group", i, sep = "")
		cat("\t",'<tmrcaStatistic id=','"','tmrca(',group,')','"','>',"\n", sep="")
		cat("\t\t",'<mrca>\n', sep="")
		cat("\t\t\t",'<taxa idref=','"',group,'"','/>',"\n", sep="")
		cat("\t\t",'</mrca>\n', sep="")
		cat("\t\t",'<treeModel idref=','"','treeModel','"','/>',"\n", sep="")
		cat("\t",'</tmrcaStatistic>\n', sep="")
		cat("\t",'<monophylyStatistic id=','"','monophyly(',group,')','"','>',"\n", sep="")
		cat("\t\t",'<mrca>\n', sep="")
		cat("\t\t\t",'<taxa idref=','"',group,'"','/>',"\n", sep="")
		cat("\t\t",'</mrca>\n', sep="")
		cat("\t\t",'<treeModel idref=','"','treeModel','"','/>',"\n", sep="")
		cat("\t",'</monophylyStatistic>\n', sep="")	}
cat("\n")
cat('\t<!-- Constraints - End Monophyly statistics -->\n')
cat("\n")	

cat('\t<!-- Define operators                                                        -->\n')
cat('\t<operators id="operators">\n')
cat('\t\t<subtreeSlide size="100.0" gaussian="true" weight="15">\n')
cat('\t\t\t<treeModel idref="treeModel"/>\n')
cat('\t\t</subtreeSlide>\n')
cat('\t\t<narrowExchange weight="15">\n')
cat('\t\t\t<treeModel idref="treeModel"/>\n')
cat('\t\t</narrowExchange>\n')
cat('\t\t<wideExchange weight="3">\n')
cat('\t\t\t<treeModel idref="treeModel"/>\n')
cat('\t\t</wideExchange>\n')
cat('\t\t<wilsonBalding weight="3">\n')
cat('\t\t\t<treeModel idref="treeModel"/>\n')
cat('\t\t</wilsonBalding>\n')
cat('\t\t<scaleOperator scaleFactor="0.75" weight="3">\n')
cat('\t\t\t<parameter idref="treeModel.rootHeight"/>\n')
cat('\t\t</scaleOperator>\n')
cat('\t\t<uniformOperator weight="30">\n')
cat('\t\t\t<parameter idref="treeModel.internalNodeHeights"/>\n')
cat('\t\t</uniformOperator>\n')
cat('\t\t<scaleOperator scaleFactor="0.75" weight="3">\n')
cat('\t\t\t<parameter idref="birthDeath.meanGrowthRate"/>\n')
cat('\t\t</scaleOperator>\n')
cat('\t\t<scaleOperator scaleFactor="0.75" weight="3">\n')
cat('\t\t\t<parameter idref="birthDeath.relativeDeathRate"/>\n')
cat('\t\t</scaleOperator>\n')
cat('\t\t<upDownOperator scaleFactor="0.75" weight="3">\n')
cat('\t\t\t<up>\n')
cat('\t\t\t</up>\n')
cat('\t\t\t<down>\n')
cat('\t\t\t\t<parameter idref="treeModel.allInternalNodeHeights"/>\n')
cat('\t\t\t</down>\n')
cat('\t\t</upDownOperator>\n')
cat('\t\t</operators>\n')
cat('\t<!-- Define MCMC                                                             -->\n')
cat('\t<mcmc id="mcmc" chainLength="11111000" autoOptimize="true">\n')
cat('\t\t<posterior id="posterior">\n')
cat('\t\t\t<prior id="prior">\n')
cat('\t\t\t\t<gammaPrior shape="0.05" scale="40.0" offset="0.0">\n')
cat('\t\t\t\t\t<parameter idref="kappa"/>\n')
cat('\t\t\t\t</gammaPrior>\n')
cat('\t\t\t\t<speciationLikelihood idref="speciation"/>\n')

cat("\n")
cat('\t<!-- Constraints - Begin Boolean Likelihood -->\n')
cat("\n")
	
cat("\t\t\t\t",'<booleanLikelihood>\n', sep="")
for (i in 1:length(clade.list)) {
		group <- paste("group", i, sep = "")
		cat("\t\t\t\t\t",'<monophylyStatistic idref=','"','monophyly(',group,')','"','/>',"\n", sep="")
    }
cat("\t\t\t\t",'</booleanLikelihood>\n', sep="")
cat('\t<!-- Node Age Priors                                                         -->\n')
for (i in 1:length(clade.list)) {
		group <- paste("group", i, sep = "")
		cat("\t\t\t\t",'<uniformPrior lower=','"',branching.times(phy)[i]-error,'"',' upper=','"',branching.times(phy)[i]+error,'"','>',"\n",sep="")
		cat("\t\t\t\t\t",'<tmrcaStatistic idref=','"','tmrca(',group,')','"','/>',"\n",sep="")
		cat("\t\t\t\t",'</uniformPrior>',"\n",sep="")
	}
	
cat("\n")
cat('\t<!-- Constraints - End Boolean Likelihood -->\n')
cat("\n")
	
cat('\t\t\t</prior>\n')
cat('\t\t\t<likelihood id="likelihood">\n')
cat('\t\t\t\t<treeLikelihood idref="treeLikelihood"/>\n')
cat('\t\t\t</likelihood>\n')
cat('\t\t</posterior>\n')
cat('\t\t<operators idref="operators"/>\n')
cat('\t\t<!-- write log to screen                                                     -->\n')
cat('\t\t<log id="screenLog" logEvery="1000">\n')
cat('\t\t\t<column label="Posterior" dp="4" width="12">\n')
cat('\t\t\t\t<posterior idref="posterior"/>\n')
cat('\t\t\t</column>\n')
cat('\t\t\t<column label="Prior" dp="4" width="12">\n')
cat('\t\t\t\t<prior idref="prior"/>\n')
cat('\t\t\t</column>\n')
cat('\t\t\t<column label="Likelihood" dp="4" width="12">\n')
cat('\t\t\t\t<likelihood idref="likelihood"/>\n')
cat('\t\t\t</column>\n')
cat('\t\t\t<column label="rootHeight" sf="6" width="12">\n')
cat('\t\t\t\t<parameter idref="treeModel.rootHeight"/>\n')
cat('\t\t\t</column>\n')
cat('\t\t\t<column label="clock.rate" sf="6" width="12">\n')
cat('\t\t\t\t<parameter idref="clock.rate"/>\n')
cat('\t\t\t</column>\n')
cat('\t\t</log>\n')
	
cat('\t\t<log id="fileLog" logEvery="1000" fileName="',file.out,'.log">\n',sep="")
cat('\t\t\t<posterior idref="posterior"/>\n')
cat('\t\t\t<prior idref="prior"/>\n')
cat('\t\t\t<likelihood idref="likelihood"/>\n')
cat('\t\t\t<parameter idref="treeModel.rootHeight"/>\n')
cat('\t\t\t<parameter idref="birthDeath.meanGrowthRate"/>\n')
cat('\t\t\t<parameter idref="birthDeath.relativeDeathRate"/>\n')
cat('\t\t\t<parameter idref="kappa"/>\n')
cat('\t\t\t<parameter idref="frequencies"/>\n')
cat('\t\t\t<parameter idref="clock.rate"/>\n')
cat('\t\t\t<treeLikelihood idref="treeLikelihood"/>\n')
cat('\t\t\t<speciationLikelihood idref="speciation"/>\n')
cat('\t\t</log>\n')
cat('\t\t<!-- write tree log to file                                                  -->\n')
cat('\t\t<logTree id="treeFileLog" logEvery="1000" nexusFormat="true" fileName="',file.out,'.trees" sortTranslationTable="true">\n',sep="")
cat('\t\t\t<treeModel idref="treeModel"/>\n')
cat('\t\t\t<strictClockBranchRates idref="branchRates"/>\n')
cat('\t\t\t<posterior idref="posterior"/>\n')
cat('\t\t</logTree>\n')
cat('\t</mcmc>\n')
cat('\t<report>\n')
cat('\t\t<property name="timer">\n')
cat('\t\t\t<mcmc idref="mcmc"/>\n')
cat('\t\t</property>\n')
cat('\t</report>\n')
cat('</beast>\n')

sink()



sink(paste(file.out, ".nex", sep=""))
cat("#NEXUS\n")
cat("\n")
cat("Begin DATA;\n")
cat("Dimensions ntax=",length(phy$tip.label)," nchar=1;\n", sep="")
cat("Format datatype=NUCLEOTIDE gap=-;\n")
cat("Matrix\n")

for (i in 1:length(phy$tip.label)) {
	cat(phy$tip.label[i],"\tN\n")
	}
cat(";\n")
cat("end;\n\n")
cat("Begin Trees;\n")
cat("\tTree StartTree = [&R] ",write.tree(phy),"\n",sep="")
cat("end;\n")
sink()

}


# example tree
tree <- read.tree("tree_filtered_with_length_all_ti.nwk")
unlink("tree_filtered_with_length_all_ti.nwk")